onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -124,14 +124,22 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
   const T* x_data = X->template Data<T>();
   int64_t zindex = 0;
 
+  std::vector<float> scores;
+  std::vector<float> kernels;
+  std::vector<int64_t> votes;
+  std::vector<float> probsp2;
+
+  const int64_t class_count_squared = class_count_ * class_count_;
+  probsp2.reserve(class_count_squared);
+  scores.reserve(class_count_squared);
+
   for (int64_t n = 0; n < N; n++)  //for each example
   {
+    scores.clear();
+    kernels.clear();
+
     int64_t current_weight_0 = n * stride;
     int64_t maxclass = -1;
-    std::vector<float> decisions;
```

Comment:
Should we have some static analysis tool added in the build process which tells us about unused variables, unreachable code, etc.? Or use flags when compiling the code which throws warnings(treated as error in the build process), for example: g++ -Wall <filename>

Reply:
It's possible to run Visual Studio code analysis on the project to detect such things. That is enabled by default for the common and graph projects but we've never invested the time in going beyond that. You can manually run it from Visual Studio though (right-click project in Solution Explorer, Analyze->Run Code Analysis on...).

Unfortunately you can't make the warnings errors by default. The code analysis rules and implementation are somewhat fluid, and depending on the toolset installed you might get different warnings. e.g. a false positive may be fixed in a newer version.

As for making warnings result in errors on Linux, that could be more achievable. Last time I looked though there there were still a lot of Linux only warnings that would need to be cleaned up. Most people develop on Windows so don't seen these warnings.